lib/pieces/white-pawn.js

```diff
@@ -7,4 +7,34 @@ function WhitePawn (square, color) {
 
 WhitePawn.prototype = Object.create(Piece.prototype);
 
+WhitePawn.prototype.canMoveTo = function (targetSquare) {
+  return canMoveForward(this, targetSquare) || canTakeDiagonally(this, targetSquare);
+};
+
+function canMoveForward (pawn, targetSquare) {
+  return ((!targetSquare.isOccupied())
+    && (pawn.square.xCoordinate === targetSquare.xCoordinate)
+    && (pawn.square.yCoordinate -1 === targetSquare.yCoordinate ))
+    || firstMoveDouble (pawn, targetSquare);
+}
+
+function firstMoveDouble (pawn, targetSquare) {
+  var firstSquareX = targetSquare.xCoordinate;
+  var firstSquareY = targetSquare.yCoordinate + 1;
+  var firstSquareIndex = (firstSquareY * 8) + firstSquareX;
+  var firstSquare = targetSquare.board.squares[firstSquareIndex];
+
+  return (pawn.moveCount === 0)
+  && (pawn.square.xCoordinate === targetSquare.xCoordinate)
+  && (pawn.square.yCoordinate -2 === targetSquare.yCoordinate)
+  && (!firstSquare.isOccupied());
```

Comment:
I don't understand how this return statement is working...

return (true) && (true) && (true) && (true)??

Does that end up just returning true at the end?